resources/config/services.yml

```diff
@@ -37,15 +37,16 @@ services:
                 logger: '@grumphp.logger'
                 inherit_environment_variables: true
 
-    grumphp.io:
-        alias: grumphp.io.console
+    GrumPHP\IO\IOFactory:
+        arguments:
+            - '@OndraM\CiDetector\CiDetector'
 
-    grumphp.io.console:
-        class: GrumPHP\IO\ConsoleIO
+    grumphp.io:
+        class: GrumPHP\IO\IOInterface
+        factory: ['@GrumPHP\IO\IOFactory', 'create']
```

Comment:
Since this service is being build from the factory - wouldnt it make sense to inject the `IOInterface` directly instead of the IOFactory?

Reply:
Suppose we could 🤔 I think the reason I injected the factory is because the command previously built the class it's self using the provided `$input, $output` args